test/lib/ansible_test/_data/sanity/validate-modules/validate_modules/main.py

```diff
@@ -1212,6 +1212,14 @@ def _validate_argument_spec(self, docs, spec, kwargs, context=None):
                     code='parameter-alias-repeated',
                     msg=msg
                 )
+            data_choices = data.get('choices', set())
+            if not context and arg == 'state':
+                bad_states = {'list', 'info'} & set(data_choices)
```

Comment:
After the refactoring, it's probably okay to inline it back since it's used in one place now:
```suggestion
                bad_states = {'list', 'info'} & set(data.get('choices', set()))
```